code/modules/mob/living/carbon/xenomorph/castes/puppeteer/abilities_puppeteer.dm

```diff
@@ -0,0 +1,449 @@
+// ***************************************
+// *********** Flay
+// ***************************************
+/datum/action/xeno_action/activable/flay
+	name = "Flay"
+	action_icon_state = "flay"
+	desc = "Takes a chunk of flesh from the victim marine through a quick swiping motion, adding 100 biomass to your biomass collection."
+	ability_name = "flay"
+	plasma_cost = 0
+	cooldown_timer = 20 SECONDS
+	target_flags = XABB_MOB_TARGET
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_FLAY,
+	)
+
+/datum/action/xeno_action/activable/flay/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	if(!ishuman(target))
+		if(!silent)
+			to_chat(owner, span_xenowarning("We cant find any suitable flesh on this thing!"))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	var/mob/living/carbon/human/target_human = target
+	if(!owner_xeno.Adjacent(target_human))
+		if(!silent)
+			to_chat(owner_xeno, span_xenonotice("We need to be next to our victim."))
+		return FALSE
+
+	if(target_human.stat == DEAD)
+		to_chat(owner_xeno, span_xenonotice("Dead meat is bad meat!"))
+		return FALSE
+
+	if(!.)
+		return
```

Comment:
why is this so randomly placed please place it according to the usual convention

Reply:
i didnt read shit about the convention